engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -3853,7 +3856,7 @@ private void handlePowerOffReportWithNoPendingJobsOnVM(final VMInstanceVO vm) {
         case Stopped:
         case Migrating:
             s_logger.info("VM " + vm.getInstanceName() + " is at " + vm.getState() + " and we received a power-off report while there is no pending jobs on it");
-            if(vm.isHaEnabled() && vm.getState() == State.Running && vm.getHypervisorType() != HypervisorType.VMware && vm.getHypervisorType() != HypervisorType.Hyperv) {
+            if(vm.isHaEnabled() && vm.getState() == State.Running && HaVmRestartHostUp.value() && vm.getHypervisorType() != HypervisorType.VMware && vm.getHypervisorType() != HypervisorType.Hyperv) {
```

Comment:
@Slair1 would you mind extracting these conditions to a method (e.g. `shouldRestartVirtualMachine`)? With your addition the if condition will have 5 elements, which is a lot. Extracting this bit to a method enabled documentation and unit test cases for these different cases.

Reply:
Yea, that sounds like a good idea, i am booked up here for a while, but it sounds like a good idea